### PR TITLE
feature resnet classifier

### PR DESCRIPTION
**Pull request by soma0sd**
# ResNet 분류기 추가

## 테스트 사전 준비

- [Chest X-Ray Images](https://www.kaggle.com/datasets/paultimothymooney/chest-xray-pneumonia)

의 학습 데이터를 Tango의  Data Management에 등록하여 사용 (Type: Image, Purpose: Classfication, Labels: PNEUMONIA / NORMAL)

## 시험

1. 새 프로젝트 생성
2. 이름과 설명 정의
3. 데이터셋은 이전에 생성한 Chest X-Ray Images 선택
4. Target 선택
5. 설정에서 Task Type을 Classification으로 선택
6. BMS -> Auto NN -> Image Deploy 순으로 실행

## 현재 이슈

- 사전학습된 가중치 파일 `kagglecxr_resnet152_normalize.pt`는 100MB 초과 바이너리(약 666MB)이기 때문에 GitHub에서 업로드를 거부함: 이 사전 학습한  가중치 파일은 `autonn/ResNet/resnet_core/pretrained/`에 위치

- `project_manager/tango/projectHandler.py`
  - `get_container_info(host_name)`: 프로젝트 매니저의 container_start, status_request시 "container=autonn_resnet"가 매개변수로 들어오는 일이 발생하며, 해당 컨테이너에 대해 "autonn-resnet" 출력이 발생해야 함
  - `get_log_container_name(container)`: 의 "container=autonn_resnet"입력이 발생. 예상하던 "autonn-resnet"입력으로 들어오지 않음

### CODE REVIEW

File: autonn/ResNet/resnet_core/resnet_utils/train.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+    with open(proj_path / "basemodel.yaml", "r") as f:
+        basemodel_info: dict = yaml.safe_load(f)
+
+    layers: List[int] = basemodel_info.get("layers", [3, 3, 3])
+    num_classes: int = basemodel_info.get("num_classes", 2)
+    input_size: int = DEFAULT_OPTIONS.get("input_size", [256])[0]
+    input_mean: float = DEFAULT_OPTIONS.get("input_mean", [0.5])[0]
+    input_std: float = DEFAULT_OPTIONS.get("input_std", [0.5])[0]
+    batch_size: int = DEFAULT_OPTIONS.get("batch_size", [4])[0]
```
Comment (iksooman) [inline autonn/ResNet/resnet_core/resnet_utils/train.py:121]: `batch size`는 `project_info.yaml`에서 읽어서 사용하는 방식으로 변경해야할 것 같습니다.

File: autonn/ResNet/resnet_core/resnet_utils/models/resnet_cifar10.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    model = ResNet(BasicBlock, [9, 9, 9], num_classes)
+    return model
+
+
+def resnet110(num_classes):
+    model = ResNet(BasicBlock, [18, 18, 18], num_classes)
+    return model
+
+
+def resnet152(num_classes):
```
Comment (iksooman) [inline autonn/ResNet/resnet_core/resnet_utils/models/resnet_cifar10.py:150]: `resnet152`는 ImageNet용 모델인데 cifar용 모델로 구현한 것 같습니다. 사전조율된 내용이면 무시하셔도 됩니다.

File: autonn/ResNet/resnet_core/resnet_utils/train.py (hunk 1 above)
Comment (soma0sd) [reply autonn/ResNet/resnet_core/resnet_utils/train.py:121]: 프로젝트 공유 디렉토리의 project_info.yaml에 `batch_size` 값이 존재하지 않아 원인을 파악하는 중입니다. 원인을 파악 후 문제상황이나 수정사항을 공유하도록 하겠습니다.

File: autonn/ResNet/resnet_core/resnet_utils/models/resnet_cifar10.py (hunk 2 above)
Comment (soma0sd) [reply autonn/ResNet/resnet_core/resnet_utils/models/resnet_cifar10.py:150]: 레이블을 resnet152로 하고 있지만 모델 구조는 cifar10을 따르고 있습니다. 오해가 없도록 설명할 수 있는 방법을 강구하겠습니다.

File: autonn/ResNet/resnet_core/resnet_utils/train.py (hunk 1 above)
Comment (iksooman) [reply autonn/ResNet/resnet_core/resnet_utils/train.py:121]: `batch_size`는 BMS 모듈에서 작성해주는 내용인데 classification쪽은 아직 구현되지 않았습니다. yaml의 `batch_size`를 읽어본 뒤에 값이 없으면 현재 default 값으로 쓰고 계신 것을 사용하시면 테스트 가능하실 것 같습니다.

File: autonn/ResNet/resnet_core/resnet_utils/models/resnet_cifar10.py (hunk 2 above)
Comment (iksooman) [reply autonn/ResNet/resnet_core/resnet_utils/models/resnet_cifar10.py:150]: 의도된 구현이면 따로 설명이 없어도 될 것 같습니다. 감사합니다.

**Comment by ML-TANGO** [comment]
https://github.com/ML-TANGO/TANGO/pull/92 에서 project manager쪽 코드 수정했다고 합니다.
확인해보시기 바랍니다.

**Review by ML-TANGO — APPROVED** [review]
PR 요청자(하늘소프트)의 테스트 의견에 따라 승인 및 merge
